src/Http/Routing/src/EndpointMiddleware.cs

```diff
@@ -11,6 +14,9 @@ namespace Microsoft.AspNetCore.Routing
 {
     internal sealed class EndpointMiddleware
     {
+        internal const string AuthorizationMiddlewareInvokedKey = "__AuthorizationMiddlewareInvoked";
```

Comment:
I wonder if these should be features... Then we could back them by fields in Kestrel and avoid the dictionary look up altogether

Reply:
> Is it possible to use mvc without ever initilizing Items?

MVC uses it in a couple of places (https://github.com/aspnet/AspNetCore/blob/c565386a3ed135560bc2e9017aa54a950b4e35dd/src/Mvc/Mvc.Core/src/Routing/UrlHelperFactory.cs, https://github.com/aspnet/AspNetCore/blob/c565386a3ed135560bc2e9017aa54a950b4e35dd/src/Mvc/Mvc.ViewFeatures/src/Filters/SaveTempDataFilter.cs#L34) so generally you'd see it initialized if MVC is involved.